packages/react-components/react-combobox/src/components/Listbox/useListbox.ts

```diff
@@ -90,6 +90,44 @@ export const useListbox_unstable = (props: ListboxProps, ref: React.Ref<HTMLElem
     }
   };
 
+  const onFocus = (_event: React.FocusEvent<HTMLElement>) => {
+    if (hasParentActiveDescendantContext || activeDescendantController.active()) {
+      return;
+    }
+
+    // restore focus to last active option (if it still exists) - similar to memorizeCurrent in useArrowNavigationGroup
+    const lastActiveDescendent = lastActiveDescendentId.current
```

Comment:
IMO this should be build into the activeDescendantController - I don't think a control specifically should deal with that just like our controls don't deal with `memorizeCurrent`.

I would propose that you memorize this state in the controller and expose a new API like `focusLastActive` to do this?

Reply:
I love this! Will add this now

Reply:
Done! Let me know what you think of how I implemented it